### ISSUE TITLE
What is the order of hosts in output of `siac hostdb -v`?

### ISSUE THREAD

**Issue by starius** [NebulousLabs/Sia#1976]
I assumed the command sorts active hosts by score, but I see discrepancies in the actual output:

```
  49:   ed25519:59585373c861444cd5d3eff576085b06f13a120f2bbf91706854ee0e0b2dfe08  zornet.zapto.org:9982                            1.84267e+07  50 SC 
  48:   ed25519:5106fc97c3c445d86acd72c83516886078cdf26a4bb13f829771182695c8c9cb  86.90.54.227:9982                                1.88213e+07  50 SC 
  47:   ed25519:de20b26c8e89b2983ecc8a81749c30c161e2237d3863b048513efcb421e22797  178.32.220.148:9982                               1.9244e+07  100 SC
  46:   ed25519:f8d4d336ac91cdad92f1e150e2298d517375513d4b3f64d87069d4d118bbfed4  212.77.177.47:9982                               2.00126e+07  200 SC
  45:   ed25519:3cd1a2e643fa026e871b126f6e927122f120071962d3b18bb38dba4437ab62c9  80.99.198.140:9982                               2.02802e+07  99 SC 
  44:   ed25519:e54208fdb47d40894f233687fb02de975098bb719b47e20bbf96938ad1ea117d  US-SIA-HOST.mooo.com:9982                        4.16207e+07  200 SC
  43:   ed25519:6c8f4dbd0561bf33072658a3548cc1a400716bd636ee8745bd85c709b31bd201  70.115.236.251:9000                              2.09829e+07  150 SC
  42:   ed25519:d9d01b722fb0ea50de9aac3950fc43308243ac2eb840d794f32dfbe7d13f0ea7  176.9.143.138:9982                               2.18645e+07  50 SC 
  41:   ed25519:1fe353fd059d06333655e3f3f7a74a428c9cda11cf5e1832355a075e3e42fb32  176.182.186.235:9980                             2.29279e+07  49 SC 
  40:   ed25519:057915ae957b294c9a367b18f2b5413f259d8e8b56508ab2b7f7ce886536e6a8  sia2.ink-online.ru:9982                          2.45364e+07  300 SC
```

**Comment by lukechampine** [comment]
They are internally sorted by "weight" ([see here](https://github.com/NebulousLabs/Sia/blob/master/modules/renter/hostdb/hosttree/hosttree.go#L197)) which is not quite the same as the "score," which is computed [here](https://github.com/NebulousLabs/Sia/blob/master/modules/renter/hostdb/hostweight.go#L411).
They should probably be re-sorted by score in siac.

**Comment by starius** [comment]
What is the purpose of having both "weight" and "score"?

**Comment by lukechampine** [comment]
cc @DavidVorick

**Comment by tbenz9** [comment]
Tag "Documentation"

**Comment by DavidVorick** [comment]
@lukechampine I'm not quite certain what's going on in the code you linked. I thought the weight and the score were supposed to be the same thing

**Comment by pachisi456** [comment]
Did some debugging and tried to find the bug today. I found that the sorting of the tree works well so that's not the problem. As it is currently implemented siad pulls the hosts sorted from the hosttree, then iterates over the hosts and makes an API call for each host to get their scores. The function called in that API call is the same as the `WeightFunc` passed to the hosttree to weigh hosts (namely `calculateHostWeight()` in `hostweight.go`). So my assumption now is that somewhere something happens that changes the parameters of a host thus resulting in a different output of `calculateHostWeight()` which is however not being updated in the hosttree. Maybe some interaction for the `interactionWeight`? Any thoughts or ideas?

I extended the API call for all hosts to include the weight data from the hosttree so I do have some code that fixes the output and has (if you want to trust the hosttree) appropriate scores ([see here](https://github.com/pachisi456/Sia/commit/90a5c2f807d456e7a7d1211ac215f50efb148a6e)). However this doesn't fix the discrepancy between the data in the hosttree and the data returned when calling `calculateHostWeight()` at any given time. If that's not a big deal for some reason I'm obviously happy to make a pull request.

**Comment by DavidVorick** [comment]
Yeah it looks like the Score() function isn't providing the interaction data, maybe it's missing some other metrics as well.

We should fix that.
